commands/src/commands.ts

```diff
@@ -86,6 +86,58 @@ function deleteText(view: EditorView, dir: "forward" | "backward") {
 export const deleteCharBackward: Command = view => deleteText(view, "backward")
 export const deleteCharForward: Command = view => deleteText(view, "forward")
 
+// FIXME support indenting by tab, configurable indent units
+
+function space(n: number) {
+  let result = ""
+  for (let i = 0; i < n; i++) result += " "
+  return result
+}
+
+function getIndentation(state: EditorState, pos: number): number {
+  for (let f of state.behavior.get(StateExtension.indentation)) {
```

Comment:
Why? Does this go for all of the static values on `StateExtension`? (Would it make sense to continue this discussion in #67 ?)

Reply:
I think it goes for all of them. Maybe because they are not extensions, but behaviors? I think `indentation` should be `indentationGetter` or `getIndentation` or `indentationAt` and they could all have a `Behavior` suffix for clarity.

Reply:
At this point, behaviors can be considered to be a type of extension.

These are not getters—if you call them, you create an extension that holds the value you pass.

Reply:
No, they are not getters, but the behavior provided by `indentation` is »getting the indentation at a specific position«. `state.behavior.get(StateExtension.indentation)` returns an array of functions that return the indentation at a specific position.